src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/config/HmppsTierExceptionHandler.kt

```diff
@@ -0,0 +1,78 @@
+package uk.gov.justice.digital.hmpps.hmppstier.config
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus.BAD_REQUEST
+import org.springframework.http.HttpStatus.NOT_FOUND
+import org.springframework.http.ResponseEntity
+import org.springframework.http.converter.HttpMessageConversionException
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.RestControllerAdvice
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.hmppstier.dto.ErrorResponse
+import uk.gov.justice.digital.hmpps.hmppstier.service.exception.EntityNotFoundException
+
+@RestControllerAdvice
+class HmppsTierExceptionHandler {
+
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @ExceptionHandler(EntityNotFoundException::class)
+  fun handleEntityNotFoundException(e: EntityNotFoundException): Mono<ResponseEntity<ErrorResponse>> {
```

Comment:
It's mighty confusing that these exc handlers use Mono when we use coroutines everywhere... I suspect it's not necessary or there is a coroutine version?!

Reply:
coroutines works